src/VisualStudio/Core/Test/CodeModel/CSharp/ExternalCodePropertyTests.vb

```diff
@@ -0,0 +1,162 @@
+}
+</Code>
+
+            Await TestOverrideKind(code, EnvDTE80.vsCMOverrideKind.vsCMOverrideKindNone)
+        End Function
+
+        <WorkItem(9646, "https://github.com/dotnet/roslyn/issues/9646")>
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestOverrideKind2() As Task
+            Dim code =
+<Code>
+abstract class C
+{
+    public abstract int $$P
+    {
+        get;
+        set;
+    }
+}
+</Code>
+
+            Await TestOverrideKind(code, EnvDTE80.vsCMOverrideKind.vsCMOverrideKindAbstract)
+        End Function
+
+#End Region
+
+#Region "Parameter name tests"
+
+        <WorkItem(9646, "https://github.com/dotnet/roslyn/issues/9646")>
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestParameterNameWithEscapeCharacters() As Task
+            Dim code =
+<Code>
+class Program
+{
+    public int $$this[int x, int y]
+    {
+        get { return x * y; }
+        set { }
+    }
+}
+</Code>
+
+            Await TestAllParameterNames(code, "x", "y")
+        End Function
+
+#End Region
+
+#Region "ReadWrite tests"
+
+        <WorkItem(9646, "https://github.com/dotnet/roslyn/issues/9646")>
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestReadWrite1() As Task
+            Dim code =
+<Code>
+class C
+{
+    public int $$P
+    {
+        get
+        {
+            return default(int);
+        }
+        set
+        {
+        }
+    }
+}
+</Code>
+
+            Await TestReadWrite(code, EnvDTE80.vsCMPropertyKind.vsCMPropertyKindReadWrite)
+        End Function
+
+        <WorkItem(9646, "https://github.com/dotnet/roslyn/issues/9646")>
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestReadWrite2() As Task
+            Dim code =
+<Code>
+class C
+{
+    public int $$P
+    {
+        get
+        {
+            return default(int);
+        }
+    }
+}
+</Code>
+
+            Await TestReadWrite(code, EnvDTE80.vsCMPropertyKind.vsCMPropertyKindReadOnly)
+        End Function
+
+        <WorkItem(9646, "https://github.com/dotnet/roslyn/issues/9646")>
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestReadWrite3() As Task
+            Dim code =
+<Code>
+class C
+{
+    public int $$P
+    {
+        set
+        {
+        }
+    }
+}
+</Code>
+
+            Await TestReadWrite(code, EnvDTE80.vsCMPropertyKind.vsCMPropertyKindWriteOnly)
+        End Function
+
+        <WorkItem(9646, "https://github.com/dotnet/roslyn/issues/9646")>
+        <ConditionalWpfFact(GetType(x86)), Trait(Traits.Feature, Traits.Features.CodeModel)>
+        Public Async Function TestReadWrite4() As Task
+            Dim code =
+<Code>
+class C
+{
+    public int $$P { get; set; }
```

Comment:
Possibly, though we'd never see one in production. The unit tests are simply constructing external code properties from source. In production, they come from metadata, where expression-bodied properties don't matter.